SIPS/sip-26.md

```diff
@@ -0,0 +1,168 @@
+  for signing requests and managing accounts.
+
+- **Protocol Snaps**: Snaps that implement protocol methods that do not require
+  an account to be executed.
+
+- **RPC Router**: Native component that forwards non-signing requests to the
+  appropriate Protocol Snap or native implementation.
+
+- **Account Router**: Native component that forwards signing requests to the
+  appropriate Account Snap or native implementation.
+
+- **Account Address Resolution Snaps**: Snaps that implement the
+  `resolveAccountAddress` method to extract the account address from the
+  request object.
+
+### Account Router
+
+The Account Router exposes the [`snap_manageAccounts`][snap-manage-accs] method
+to allow account Snaps to register, remove, and update accounts.
+
+For example, this code can be used by an Account Snap to register a new account
+with the Account Router:
+
+```typescript
+// This will notify the Account Router that a new account was created, and the
+// Account Router will register this account as available for signing requests
+// using the `eth_signTypedData_v4` method.
+await snap.request({
+  method: 'snap_manageAccounts',
+  params: {
+    method: 'notify:accountCreated',
+    params: {
+      account: {
+          id: '74bb3393-f267-48ee-855a-2ba575291ab0',
+          type: 'eip155:eoa',
+          address: '0x1234567890123456789012345678901234567890',
+          methods: ['eth_signTypedData_v4'],
+          options: {},
+      },
+    },
+  },
+});
+```
+
+Similar events are available to notify about the removal and update of
+accounts: `notify:accountRemoved` and `notify:accountUpdated`.
+
+Additionally, the Account Router expects the Account Snap to implement the
+Keyring API so it can forward signing requests to it through the
+[`keyring_submitRequest`][submit-request] method.
+
+### Account Address Resolution Snaps
+
+The Account Address Resolution Snaps are responsible for extracting the account
+address that should receive the signing request from the request object. This
+is accomplished by exposing the `resolveAccountAddress` method to the Account
+Router.
+
+```typescript
+/**
+ * Returns the address of the account that should handle the signing request.
+ *
+ * @param request - The request object.
+ * @returns The account address or `undefined` if the address could not be
+ * resolved.
+ */
+function resolveAccountAddress(request: MultichainRequest): string | undefined;
+```
+
+There must be only one Account Resolution Snap registered per chain to prevent
+ambiguity in the account resolution process.
+
+To identify which Account Resolution Snap should be used for a given request,
+Account Address Resolution Snaps MUST list the supported chains in their
+manifest file using the [CAIP-2][caip-2] format for chain IDs:
+
+```json5
+"initialPermissions": {
+  "endowment:account-address-resolver": {
+    "chains": [
+      "<chain_id_1>",
+      "<chain_id_2>",
+      // ...
+    ]
+  }
```

Comment:
same as below, should resemble caip-25 permission: https://github.com/MetaMask/SIPs/pull/147/files#r1777172390